packages/cli/src/build/__tests__/imports-conditions.ts

```diff
@@ -253,6 +260,139 @@ test("imports conditions", async () => {
+  // see https://github.com/jestjs/jest/issues/7997
+  let dirPromise: Promise<string>;
+
+  beforeAll(async () => {
+    let resolve: (dir: string) => void;
+    dirPromise = new Promise((_resolve) => (resolve = _resolve));
+
+    const dir = await testdir({
+      "package.json": JSON.stringify({
+        name: "@scope/pkg",
+        main: "dist/scope-pkg.cjs.js",
+        module: "dist/scope-pkg.esm.js",
+        exports: {
+          ".": {
+            types: {
+              import: "./dist/scope-pkg.cjs.mjs",
+              default: "./dist/scope-pkg.cjs.js",
+            },
+            browser: {
+              development: {
+                module: "./dist/scope-pkg.browser.development.esm.js",
+                import: "./dist/scope-pkg.browser.development.cjs.mjs",
+                default: "./dist/scope-pkg.browser.development.cjs.js",
+              },
+              module: "./dist/scope-pkg.browser.esm.js",
+              import: "./dist/scope-pkg.browser.cjs.mjs",
+              default: "./dist/scope-pkg.browser.cjs.js",
+            },
+            development: {
+              module: "./dist/scope-pkg.development.esm.js",
+              import: "./dist/scope-pkg.development.cjs.mjs",
+              default: "./dist/scope-pkg.development.cjs.js",
+            },
+            module: "./dist/scope-pkg.esm.js",
+            import: "./dist/scope-pkg.cjs.mjs",
+            default: "./dist/scope-pkg.cjs.js",
+          },
+          "./package.json": "./package.json",
+        },
+        preconstruct: {
+          exports: {
+            importConditionDefaultExport: "default",
+          },
+          ___experimentalFlags_WILL_CHANGE_IN_PATCH: {
+            importsConditions: true,
+          },
+        },
+        imports: {
+          "#is-development": {
+            development: "./src/true.js",
+            default: "./src/false.js",
+          },
+          "#is-browser": {
+            browser: "./src/true.js",
+            default: "./src/false.js",
+          },
+        },
+      }),
+      "src/true.js": js`
+        export default true;
+      `,
+      "src/false.js": js`
+        export default false;
+      `,
+      "src/index.js": js`
+        import isDevelopment from "#is-development";
+        import isBrowser from "#is-browser";
+        if (isDevelopment) {
+          console.log("development");
+        } else {
+          console.log("production");
+        }
+        if (isBrowser) {
+          console.log("browser");
+        } else {
+          console.log("node");
+        }
+      `,
+      node_modules: typescriptFixture.node_modules,
+      "runtime-check.mjs": js`
+        import "@scope/pkg";
+      `,
+      "runtime-check.cjs": js`
+        require("@scope/pkg");
+      `,
+    });
+    await build(dir);
+
+    resolve!(dir);
+  });
+
+  describe.each(["cjs", "mjs"] as const)("with %s", (moduleType) => {
```

Comment:
usually, I hate nested describe blocks a lot but I think it makes sense here when using `.each`